llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -6970,6 +6968,122 @@ void LoopVectorizationCostModel::setCostBasedWideningDecision(ElementCount VF) {
   }
 }
 
+void LoopVectorizationCostModel::setVectorizedCallDecision(ElementCount VF) {
+  if (VF.isScalar())
```

Comment:
This function is only called from collectUniformsAndScalars for non-scalar VFs, so I think you can simply assert that the VF is scalar.